markup/asciidocext/convert.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package asciidocext converts Asciidoc to HTML using Asciidoc or Asciidoctor
+// external binaries. The `asciidoc` module is reserved for a future golang
+// implementation.
+package asciidocext
+
+import (
+	"os/exec"
+	"path/filepath"
+
+	"github.com/gohugoio/hugo/identity"
+	"github.com/gohugoio/hugo/markup/asciidocext/asciidocext_config"
+	"github.com/gohugoio/hugo/markup/converter"
+	"github.com/gohugoio/hugo/markup/internal"
+)
+
+/* ToDo: RelPermalink patch for svg posts not working*/
+type pageSubset interface {
+	RelPermalink() string
+}
+
+// Provider is the package entry point.
+var Provider converter.ProviderProvider = provider{}
```

Comment:
I see that you are using this exported variable to create a new provider, shouldn't this `var Provider` be inside a factory function in order to avoid creating an instance until it's really needed? You shouldn't have an import side-effect.

(Sorry if the question seems wrong, but I'm just learning Go.)

Reply:
The struct created is really cheap to create, so I see no win adding complexity making this more complex.